streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -4523,37 +4523,34 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   def fromIteratorSucceed[A](iterator: => Iterator[A], maxChunkSize: => Int = DefaultChunkSize)(implicit
     trace: Trace
   ): ZStream[Any, Nothing, A] = {
-
     def writeOneByOne(iterator: Iterator[A]): ZChannel[Any, Any, Any, Any, Nothing, Chunk[A], Any] =
       if (iterator.hasNext)
         ZChannel.write(Chunk.single(iterator.next())) *> writeOneByOne(iterator)
       else
         ZChannel.unit
 
-    def writeChunks(iterator: Iterator[A]): ZChannel[Any, Any, Any, Any, Nothing, Chunk[A], Any] =
```

Comment:
AFAICT the main optimization is to use `val builder = ChunkBuilder.make[A](maxChunkSize)` is that right?

If that's the case maybe keep this method and make the optimization within it just to keep blame / git history with as few changes as possible (which makes it easier to track potential bugs)

Reply:
Done! I minimized the changes